opendatasoft/semantic-bot
# Update Named Entity Recognition to use Virtuoso

THIS PR IS CLOSED FOR THE MOMENT. WE NEED TO EXPLORE OTHER SOLUTIONS SUCH AS POSTGRESQL OR ELASTICSEARCH.

In order to recognize the semantic types contained in fields of a dataset, we search n first values of each field in DBpedia dataset (Semantic version of Wikipedia). 

The deployed version of semantic bot cannot use full-text search because it use HDT files to store DBpedia (only supports exact match).

## What's new ?

Name Entity Recognition now uses Virtuoso instead of HDT files. (Open source Virtuoso)

[Virtuoso](http://vos.openlinksw.com/owiki/wiki/VOS) is a data virtualization platform. It proposes a SPARQL Endpoint that we use in the semanticbot to store and perform full-text searches on RDF data. SemanticBot use Virtuoso as it without any code modification.

We use a dockerized open source version available under [GNU General Public License (GPL) Version 2](http://vos.openlinksw.com/owiki/wiki/VOS/VOSLicense)

The architecture of the new version of the SemanticBot is available [Here](https://docs.google.com/presentation/d/1K2Ond-jOxCSRIjQmSXbxkzuZPAwE53KhMSwNYIJ7XUY/edit?usp=sharing)

## TO TEST:
Open developer tool on the navigator -> network panel
Check that no requests fail when:
- if the dataset contains a large number of fields (it will stress the db with a lot of HTTP requests)
